plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/MapToCommand.java

```diff
@@ -157,21 +159,16 @@ public void redo() {
 	}
 
 	protected FBNetworkElement createTargetElement() {
-		FBNetworkElement created = null;
-		if (srcElement instanceof ConfigurableFB) {
-			created = createConfigureableFB();
-		} else if (srcElement instanceof FB) {
-			created = createTargetFB();
-		} else if (srcElement instanceof final SubApp subApp) {
-			if (subApp.isTyped()) {
-				created = createTargetTypedSubApp();
-			} else {
-				created = createTargetUntypedSubApp();
-			}
-		}
+		final FBNetworkElement created = switch (srcElement) {
+		case final ConfigurableFB confFB -> createConfigureableFB();
+		case final FB fb -> createTargetFB();
+		case final TypedSubApp typedSubapp -> createTargetTypedSubApp();
+		case final UntypedSubApp untypedSubapp -> createTargetUntypedSubApp();
+		default -> null;
+		};
 		if (created != null) {
 			created.setName(srcElement.getName());
-			transferFBParams(srcElement, created);
+			MappingTargetCreator.transferFBParams(srcElement, created);
 		}
 		return created;
 	}
```

Comment:
Could we also replace this with `MappingTargetCreator.createMappingTarget(...)`?

Reply:
I have planned that for a later commit yes.